python/gtsam/examples/IMUKittiExampleGPS.py

```diff
@@ -0,0 +1,364 @@
+                             gps_measurements[i].position)
+            if (i % gps_skip) == 0:
+                new_factors.addPriorPose3(current_pose_key, gps_pose,
+                                          noise_model_gps)
+                new_values.insert(current_pose_key, gps_pose)
+
+                print(f"############ POSE INCLUDED AT TIME {t} ############")
+                print(gps_pose.translation(), "\n")
+            else:
+                new_values.insert(current_pose_key, current_pose_global)
+
+            # Add initial values for velocity and bias based on the previous
+            # estimates
+            new_values.insert(current_vel_key, current_velocity_global)
+            new_values.insert(current_bias_key, current_bias)
+
+            # Update solver
+            # =======================================================================
+            # We accumulate 2*GPSskip GPS measurements before updating the solver at
+            # first so that the heading becomes observable.
+            if i > (first_gps_pose + 2 * gps_skip):
+                print(f"############ NEW FACTORS AT TIME {t:.6f} ############")
+                new_factors.print()
+
+                isam.update(new_factors, new_values)
+
+                # Reset the newFactors and newValues list
+                new_factors.resize(0)
+                new_values.clear()
+
+                # Extract the result/current estimates
+                result = isam.calculateEstimate()
+
+                current_pose_global = result.atPose3(current_pose_key)
+                current_velocity_global = result.atVector(current_vel_key)
+                current_bias = result.atConstantBias(current_bias_key)
+
+                print(f"############ POSE AT TIME {t} ############")
+                current_pose_global.print()
+                print("\n")
+
+    return isam
+
+
+def main():
+    """Main runner."""
+    args = parse_args()
+    kitti_calibration, imu_measurements, gps_measurements = loadKittiData()
+
+    if not kitti_calibration.bodyTimu.equals(Pose3(), 1e-8):
+        raise ValueError(
+            "Currently only support IMUinBody is identity, i.e. IMU and body frame are the same"
+        )
+
+    # Configure different variables
+    first_gps_pose = 1
+    gps_skip = 10
+
+    # Configure noise models
+    noise_model_gps = noiseModel.Diagonal.Precisions(
+        np.asarray([0, 0, 0] + [1.0 / 0.07] * 3))
+
+    sigma_init_x = noiseModel.Diagonal.Precisions(
+        np.asarray([0, 0, 0, 1, 1, 1]))
+    sigma_init_v = noiseModel.Diagonal.Sigmas(np.ones(3) * 1000.0)
+    sigma_init_b = noiseModel.Diagonal.Sigmas(
+        np.asarray([0.1] * 3 + [5.00e-05] * 3))
+
+    isam = optimize(gps_measurements, imu_measurements, sigma_init_x,
+                    sigma_init_v, sigma_init_b, noise_model_gps,
+                    kitti_calibration, first_gps_pose, gps_skip)
+
+    save_results(isam, args.output_filename, first_gps_pose, gps_measurements)
```

Comment:
@varunagrawal is the python output the same as the c++ output?

Reply:
Here's the diff https://www.diffchecker.com/7QT5mumN